### DISCUSSION TITLE
Any way to know anchor tracking confidence?

### DISCUSSION THREAD

**Discussion by sarimmehdi** [sceneview/sceneform-android#265] [Q&A]
As we already know, if we move far away from an anchor and/ or stop looking at it, it starts drifting. Is there a way to measure the confidence with which ARCore is tracking the anchor? Like a value between 0 and 1.

So, if I move far away from the anchor I can be notified by ARCore that it is tracking the anchor with a low confidence and I take the necessary action.

**Comment by ThomasGorisse** [comment]
What version of Sceneform are you using?

**Reply by sarimmehdi** [reply]
I am actually still using the deprecated one and not this one

**Reply by ThomasGorisse** [reply]
Issue fixed in Sceneform Maintained latest versions

**Reply by sarimmehdi** [reply]
Can you tell me the name of the method that lets me know the confidence value associated to anchor tracking

**Comment by ThomasGorisse** [comment]
Answers from Discord (Guys, please continue discussion here)

_grassydragon_

> The anchor is attached to the visual features recognised by ARCore so it should return back to its original pose once you look at it. Actually, I think it is not the anchor drifting but the imprecision of measuring the change in the user pose. Unfortunately, there is no API for accessing the confidence of the anchor.

_Giovanni Giorgio_

> I guess the algorithm used for estimating the anchor pose is being kept a secret? We are working on an AR app that use anchors over a large area and we want to use as few anchors as possible and I was thinking whether there was a way to know how bad the estimation of the anchor is
> also, I was under the impression that SLAM updates the pose of the camera and surrounding landmarks (anchors in this case) simultaneously
> so, the imprecision would probably be in both the camera and anchors
> I guess Google wish to keep their underlying SLAM a secret?
> 

_grassydragon_

> Yes, ARCore has closed source code. I have read a paper about the precision of localisation in an indoor environment using AR (https://www.researchgate.net/publication/340047854_Localization_Limitations_of_ARCore_ARKit_and_Hololens_in_Dynamic_Large-scale_Industry_Environments). ARCore and ARKit didn't have very reliable results so I think you just need to do some testing and use some heuristics.

_grassydragon_

> I suppose, yes, but if you don't see the visual features that are used for positioning the anchor you can't update it 😃

_Giovanni Giorgio_

> thanks, it looks like a nice opportunity for research as well
